camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -0,0 +1,944 @@
+# coding=utf-8
+"""Tests for ``qpc source`` commands.
+
+:caseautomation: automated
+:casecomponent: cli
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+import json
+import random
+from io import BytesIO
+
+import pexpect
+import pytest
+
+from camayoc import utils
+from camayoc.constants import CONNECTION_PASSWORD_INPUT
+from camayoc.tests.qcs.cli.utils import cred_add, source_show
+
+
+NEW_HOSTS = (
+    '127.0.0.42',
+    '192.168.0.0/24',
+    '192.168.0.[1:100]'
+)
+
+
+def default_port_for_source(source_type):
+    """Resolve the default port for a given resource type."""
+    if source_type == 'vcenter':
+        return 443
+    else:
+        return 22
+
+
+@pytest.mark.parametrize('hosts', ('127.0.0.1', '192.168.0.0/24'))
+def test_add_with_cred_hosts(
```

Comment:
I like the parametrized fixture that provides the source type, but as you point out in https://github.com/quipucords/quipucords/issues/449, it is a bug if we can add a vcenter or satellite source with the CIDR notation. I'm not seeing how you are handling that in this test case, maybe I'm missing something.

Reply:
That is why I was suggesting having groupings in tuples or some other object that paired types with appropriate host values. So you could match satellite and vcenter with both alphabetical DNS names and individual IPv4 addresses, and for network do those as well as CIDER and the 10.10.182.[0:255] range notation. But there are plenty of ways you could address this.

Reply:
You are right, I am not handling the difference here yet. I found this bug later yesterday and opened the issue and updated the PR with the new code I was able to get done.

For hosts in specific I think another fixture could be created so we can group the source type with the expected hosts that it should work. Also we can create another one with invalid data to make sure we perform a negative test too.

Reply:
Nice, that would be good, and if you put the conftest.py in a couple levels up, I could probably use the fixture in the API tests as well